core/state/trie_prefetcher.go

```diff
@@ -132,41 +131,43 @@ func (p *triePrefetcher) copy() *triePrefetcher {
 		return copy
 	}
 	// Otherwise we're copying an active fetcher, retrieve the current states
-	for root, fetcher := range p.fetchers {
-		copy.fetches[root] = fetcher.peek()
+	for id, fetcher := range p.fetchers {
+		copy.fetches[id] = fetcher.peek()
 	}
 	return copy
 }
 
 // prefetch schedules a batch of trie items to prefetch.
-func (p *triePrefetcher) prefetch(root common.Hash, keys [][]byte) {
+func (p *triePrefetcher) prefetch(owner common.Hash, root common.Hash, keys [][]byte) {
 	// If the prefetcher is an inactive one, bail out
 	if p.fetches != nil {
 		return
 	}
 	// Active fetcher, schedule the retrievals
-	fetcher := p.fetchers[root]
+	id := p.trieID(owner, root)
```

Comment:
So, this is a change. Let's say we have two accounts, `A`, and `B`. Both have the same storage root `C`. 
Previously, we would spin up only one fetcher, and make all `C` lookups go to that fetcher. 
With this change, we will resolve the `C` trie twice. I suspect it doesn't matter a whole lot, just wanted to highlight it

Reply:
Yes, actually it's intended. With new state scheme, each storage trie will have its own storage space. To we need to explicitly distinguish different storage tries.